common/djangoapps/third_party_auth/utils.py

```diff
@@ -0,0 +1,43 @@
+from random import randint
+
+from openedx.core.djangoapps.site_configuration import helpers as configuration_helpers
+
+from django.contrib.auth.models import User
+
+
+class GeneratorUsername(object):
+
+    def __init__(self):
+        self.separator_character = '_'
+        self.in_lowercase = True
+        self.random = True
+
+    def separator(self, fullname):
+        username = fullname.replace(' ', self.separator_character)
+        return username
+
```

Comment:
Something weird is happening here. When I pass dot or underscore characters, works well and in the form render the username that we are generating. But with others characters like `$` or `&` it does not work with the username that we are creating and render the username that testshib generates.

Reply:
We could check this. Replace function it seems to be doing the right work here